core/src/main/java/org/kiwix/kiwixmobile/core/main/CoreMainActivity.java

```diff
@@ -1545,6 +1548,15 @@ private void updateUrlProcessor() {
     }
   }
 
+  private void updateNightMode() {
+    painter.update(getCurrentWebView(), kiwixWebView -> {
+      if (kiwixWebView.getUrl() != null) {
```

Comment:
this is too complicated, any time you see

```
if { return expression } else return true
```

it is guaranteed to be simplifiable and probably automatically. For starters it could definitely be a ternary

eg
```
return kiwixWebView.getUrl() != null ? !kiwixWebView.getUrl().equals(HOME_URL) : true
```

![image](https://user-images.githubusercontent.com/3358282/78235647-dd1e6e00-74d0-11ea-9ca4-e8eaa2805c21.png)

![image](https://user-images.githubusercontent.com/3358282/78235677-e7406c80-74d0-11ea-8331-382ae99bced8.png)

![image](https://user-images.githubusercontent.com/3358282/78235760-05a66800-74d1-11ea-9d4a-482df1cbf8f8.png)


Also you could have skipped all of these steps by pressing `alt+enter` on the `if`

![image](https://user-images.githubusercontent.com/3358282/78235925-3a1a2400-74d1-11ea-87a8-ff4953fb6ace.png)

And you could have done it before these changes by pressing `alt+enter` on the `&&`

![image](https://user-images.githubusercontent.com/3358282/78236094-777eb180-74d1-11ea-95db-ea04f369eba5.png)

Reply:
Apologies, I did something else by mistake. This is what I intended to do. I updated it now properly.